druid-shell/src/dialog.rs

```diff
@@ -21,7 +21,16 @@ use std::path::{Path, PathBuf};
 /// This path might point to a file or a directory.
 #[derive(Debug, Clone)]
 pub struct FileInfo {
+    /// The path to the selected file.
+    /// On macOS, this is already rewritten to use the extension that the user selected
+    /// with the `file format` property.
     pub(crate) path: PathBuf,
+    /// The selected file format. If there're multiple different formats available
+    /// this allows understanding the kind of format that the user expects the file
+    /// to be written in. Examples could be Blender 2.4 vs Blender 2.6 vs Blender 2.8.
+    /// The `path` above will already contain the appropriate extension chosen in the
+    /// `format` property, so it is not necessary to mutate `path` any further.
+    pub(crate) format: Option<FileSpec>,
```

Comment:
maybe we need this to be `pub`? Do we expect the user to potentially inspect it?

Reply:
But wouldn't `path` then also need to be `pub`🤔? I've changed both to be `pub` for now